src/lib.rs

```diff
@@ -310,18 +335,47 @@ impl Message {
                     match last_tag.as_str() {
                         "61" | "86" => {
                             if let Some(sl) = statement_lines.last_mut() {
-                                if let Some(ref mut info) = sl.information_to_account_owner {
-                                    info.push_str(&info_to_account_owner);
-                                } else {
-                                    sl.information_to_account_owner = Some(info_to_account_owner);
+                                match (&mut sl.information_to_account_owner, info_to_account_owner)
+                                {
+                                    (None, info) => {
+                                        sl.information_to_account_owner = Some(info);
+                                    }
+                                    (
+                                        Some(InformationToAccountOwner::Plain(sl_info)),
+                                        InformationToAccountOwner::Plain(info),
+                                    ) => {
+                                        sl_info.push_str(&info);
+                                    }
+                                    (
+                                        Some(InformationToAccountOwner::Plain(_)),
+                                        structured @ InformationToAccountOwner::Structured {
+                                            ..
+                                        },
+                                    ) => {
+                                        sl.information_to_account_owner = Some(structured);
+                                    }
+                                    _ => {}
```

Comment:
I find this (and the one below) a bit painful to read. I don't really have a suggestion to improve this match. Could you try to golf it a bit or at least add some comments what these cases mean semantically?